libs/unity/library/Runtime/Scripts/Media/MicrophoneSource.cs

```diff
@@ -75,6 +107,47 @@ protected async void OnEnable()
             AudioSourceStarted.Invoke(this);
         }
 
+#if PLATFORM_ANDROID
+        protected void OnApplicationFocus(bool hasFocus)
+        {
+            if (!hasFocus)
+            {
+                return;
+            }
+
+            // If focus is restored after a pending request, check the permission again
+            if (_androidRecordAudioRequestPending)
+            {
+                _androidRecordAudioRequestPending = false;
+
+                if (Permission.HasUserAuthorizedPermission(Permission.Microphone))
+                {
+                    // If now authorized, start capture as if just enabled
+                    Debug.Log("User granted authorization to access microphone, starting MicrophoneSource now...");
+                    OnEnable();
+                }
+                else if (Time.time <= _androidRecordAudioRequestRetryUntilTime)
```

Comment:
It seems that if we take this branch then we'll get stuck (won't enable the source nor log the error message) until the application loses focus and gets it again, is that correct? Rather than relying on focus, should we check every e.g. 1s with a coroutine?

Reply:
For how long do you suggest to check? There is no way to tell if the user already denied access and there's no dialog, or if it is the first time and the dialog is still open and the user didn't select an option yet; Java has the distinction but Unity doesn't expose it and return the same result in both cases. The latter can take an indeterminate amount of time, and having a timeout on a user dialog is pretty bad. But if it's the former, then you will continue to check forever every second for no reason, which is almost worse.

With the current approach:
- If the user already denied, we never check again. Granted we lose the error message; we could maybe have a warning when we try the first time in `OnEnable()` just for debugging. But at least we likely will never check again, which is what should happen since there's no chance we'll get permission.
- If the dialog is open, then everything works as intended since the focus will change and we'll get the timeout.

Reply:
> For how long do you suggest to check?

300s? 😄 

Let me re-frame the problem. I agree everything works fine if the user accepts the dialog and permission is granted (first branch here). If the user waits 5 min and then denies permission, it also works fine (third branch here).

If the user denies permission before 5 min (second branch), capture silently fails. An error message is printed only when (if) a focus switch happens after 5 min. Also the component being disabled or not depends on when the focus switch happens. So the behavior can be improved here.

I'd say we should:
- keep checking for permission granted (and only for that) in `OnApplicationFocus`
- check on a 5min timeout in a coroutine started from `OnEnable` rather than in `OnApplicationFocus`, so whether a denial is detected doesn't depend on focus switches
- avoid disabling the component on permission denied? Since we can't reliably detect denial I'd avoid influencing the app behavior.

Reply:
`MicrophoneSource` and `WebcamSource` always disable themselves if failing to create a track, as we tie the track lifetime to the enable/disable state. Failing to do so only in one specific case and only on Android will cause confusion.

Reply:
(well at least after #436 which I was about to open)

Reply:
> only in one specific case and only on Android

Of course we'd need to adopt the same logic on UWP too. And I'd avoid disabling the component on failure to create a track too, for the reason below.

> we tie the track lifetime to the enable/disable state

Not really - starting the track is asynchronous so there is inevitably a lapse of time where the component is enabled but there is no track. Users should use `IsStreaming`/`XStreamStarted` to detect whether there is an active track. My proposal is that the component would stay in that state forever on failure, which is less surprising IMHO than staying there for a few seconds or for 5 min and then automatically reverting to disabled. But I guess this might be subjective.

Reply:
Alright. I guess the argument is also that we need to stick to `OnEnable()` for creating the track to make sure that if the track exists then the `Update()` method is called, so a separate public `StartCaptureAsync()` is too dangerous for that reason.

One thing bothering me though : what happen if you enable then disable then enable quickly, so that the first track has not yet finished being created. I suspect that because of `await` you end up trying to create a second track while the first is not done yet unfortunately, so we cannot just rely on `OnDisable()` always occurring after the track is created, nor `OnEnable()` after the track was destroyed, and need more synchronizing.

Reply:
Logged as #439 to look at that issue specifically.